### PR TITLE
fix(932100): update data list

### LINKED ISSUE #2673: update word list for rule 932100 (RCE Unix command injection part 1/4)

### PR DESCRIPTION
**Pull request by fzipi**
Signed-off-by: Felipe Zipitria <felipe.zipitria@owasp.org>

Updated data list based on unix-shell.data.

I've also added some regexes for programming languages with versions.

Fixes #2673 .

### CODE REVIEW

**Comment by fzipi** [comment]
- Added more tests
- Changed test titles to match our generic naming

**Comment by RedXanadu** [comment]
Sorry, I haven't had time to look at this one either, yet. It, also, is still on my todo list.

**Comment by dune73** [comment]
I think the following items would profit from `@`.

```
ash
bridge
bundler ...
column
dd
docker
finger
genie
gimp
install
latex
```

**Review by theseion — CHANGES_REQUESTED** [review]
\+ things spotted by @dune73

**Comment by fzipi** [comment]
Addressed comments, and added more bare English words with `@`. Also, removed some repeated ones.

**Review by theseion — APPROVED** [review]

**Comment by RedXanadu** [comment]
Oops, sorry: I didn't follow up my earlier comment. After looking at this more closely, the same as with #2677: I don't think there are concerns about adding in anti-evasion patterns as there isn't a specific suffix being searched for, so it's fine (e.g. `lua5"".4` won't work as an evasion as it will still match on `lua`.)

**Comment by fzipi** [comment]
In summary: adding the numbers won't give us added value in this case, so I'm reverting them back to their original positions in the file and removing the specific cases.

File: util/regexp-assemble/data/932100.data [hunk 1]
```diff
@@ -127,116 +149,170 @@ bzless
...
 function
+gawk
 gcc
-gcc@
+gcore
 gdb
 gem@
+genie@
+genisoimage
+GET@
```
Comment (theseion) [inline util/regexp-assemble/data/932100.data:258]: Wasn't `GET` removed earlier??

Comment (theseion) [reply util/regexp-assemble/data/932100.data:258]: Ah, it was moved from the top.

**Review by theseion — APPROVED** [review]
LGTM

**Comment by lifeforms** [comment]
Tests are failing now, but that could be due to a temporary problem with our pipeline that should be solved now. Re-running jobs.

**Comment by lifeforms** [comment]
Two test failures persist:

running 932100-28: 💥 failed in 4.404215ms - POST data: `arg=;gcc10.1`
running 932100-29: 💥 failed in 4.240415ms - POST data: `arg=;irb31`

**Comment by fzipi** [comment]
That happens after the last update from @RedXanadu. I've removed the suffix search. But looks like now we are not matching anymore.

**Comment by theseion** [comment]
Note that this issue is affected by the same thing as #2677. We need to figure out how we ignore `python` but still catch `python3`. I've suggested adding an additional symbol for an adjusted anti-evasion pattern in the other PR.

**Comment by fzipi** [comment]
Updated:
- `irb` -> `irb~`
- `gcc` -> `gcc~`

**Comment by fzipi** [comment]
🤔 Looks like they fail when using `~`... but didn't failed previously. Let's try removing that.

**Comment by theseion** [comment]
I might be stating the obvious but you'll need to change `gcc` to `gcc~` in the data file for the new test to work (same for `irb`).

**Comment by fzipi** [comment]
Thanks for the tips. Looks like working now. Following approval, let's merge then.

**Pull request merged**